### DISCUSSION TITLE
Cumbersome to conditionally pass Value.absent()

### DISCUSSION THREAD

**Discussion by fgblomqvist** [simolus3/drift#1209] [Q&A] [answered]
If I have an optional parameter `int? foo` that I want to pass to a certain model field `foo` (let's say, an integer field with a default value), it gets quite wordy:
```dart
ThingCompanion.insert(
    ...
    foo: foo == null ? const Value.absent() : Value(startingBalance)
)
```
Is there a better way to do this? I feel like it's a quite common scenario.

**Comment by simolus3** [comment, marked as answer]
Thank you for starting this discussion. I've added `Value.ofNullable` which is similar, except that it checks that `T` is non-nullable as well:

```
Value<int>.ofNullable(null) -> absent
Value<int>.offNullable(12) -> present
Value<int?>.ofNullable(null) -> invalid (assertion error)! the null value could be both absent and present
Value<int?>.ofNullable(12) -> present
```

Until that is released, you can use your own function for that:

```dart
Value<T> valueOf<T>(T? value) {
  assert(value != null || null is! T);
  return value != null ? Value(value) : const Value<Never>.absent()`
}
```

**Reply by fgblomqvist** [reply]
Nice, looks good! Thanks 😄
